src/easymde.rs

```diff
@@ -0,0 +1,106 @@
+use std::{borrow::Cow, path::PathBuf};
+
+use axum::{extract::multipart::MultipartError, http::StatusCode, response::IntoResponse, Json};
+use serde::Serialize;
+use tracing::error;
+
+#[derive(Debug, Clone)]
+pub struct EditorConfig {
+    /// enable drag-and-drop upload functionality in the default markdown editor
+    pub(crate) enable_uploads: bool,
+    /// max upload size in bytes
+    pub(crate) upload_max_size: u32,
+    /// Allowed file types to upload. Default: image/png, image/jpeg
+    pub(crate) allowed_file_types: Vec<Cow<'static, str>>,
+}
+
+impl Default for EditorConfig {
+    fn default() -> Self {
+        Self {
+            enable_uploads: true,
+            upload_max_size: 1024 * 1024 * 2,
+            allowed_file_types: vec!["image/png".into(), "image/jpeg".into()],
+        }
+    }
+}
+
+impl EditorConfig {
+    /// Enable uploads directly in the editor.
+    pub fn enable_uploads(mut self, enable: bool) -> Self {
+        self.enable_uploads = enable;
+        self
+    }
+
+    /// Set the max size for uploads.
+    pub fn upload_max_size(mut self, max_size: u32) -> Self {
+        self.upload_max_size = max_size;
+        self
+    }
+
+    /// Add an allowed file type to the currently allowed file types.
+    pub fn allow_file_type(mut self, file_type: impl Into<Cow<'static, str>>) -> Self {
+        self.allowed_file_types.push(file_type.into());
+        self
+    }
+
+    /// Reset the allowed file types to the given list.
+    pub fn allowed_file_types(mut self, file_types: Vec<Cow<'static, str>>) -> Self {
+        self.allowed_file_types = file_types;
+        self
+    }
+}
```

Comment:
This looks like a good use case for e. g. [`getset`](https://docs.rs/getset/latest/getset/)